pkg/rc/farm.go

```diff
@@ -65,22 +67,27 @@ func NewFarm(
 	logger logging.Logger,
 	rcSelector klabels.Selector,
 	alerter alerting.Alerter,
+	errorReporter error_reporter.Reporter,
```

Comment:
I don't see any precedent for that style of injection. I don't mind the extra nil in constructors fwiw